### ISSUE TITLE
Can't process received frame

### ISSUE THREAD

**Issue by swex** [denpamusic/PyPlumIO#16]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### I'm having the following issue:

not a real issue, but looks like some frames have *another* checksum algo or something. I think its so, because of pattern in bad checksums... see logs.

### My environment is:

```markdown
- OS: Ubuntu 22.04
- Python: 3.11
```

### I have following devices connected:

- [X] ecoMAX 3xx series
- [ ] ecoMAX 8xx series
- [ ] ecoMAX 9xx series
- [ ] Expansion module B
- [ ] Expansion module C
- [ ] ecoSTER 200/ecoSTER Touch
- [ ] ecoLAMBDA
- [ ] ecoNET 300

### I'm connecting to my devices using:

USB to RS-485 adapter

### I'm seeing following log messages:

```shell
Sep 25 14:10:26  hass[1654340]: 2023-09-25 14:10:26.024 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (0)
Sep 25 14:13:09  hass[1654340]: 2023-09-25 14:13:09.099 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (22)
Sep 25 14:16:59  hass[1654340]: 2023-09-25 14:16:59.590 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (0)
Sep 25 14:22:49  hass[1654340]: 2023-09-25 14:22:49.956 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (10)
Sep 25 14:32:56  hass[1654340]: 2023-09-25 14:32:56.410 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (22)
Sep 25 14:41:53  hass[1654340]: 2023-09-25 14:41:53.140 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (104)
Sep 25 15:00:16  hass[1654340]: 2023-09-25 15:00:16.113 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (22)
Sep 25 15:01:37  hass[1654340]: 2023-09-25 15:01:37.875 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (104)
Sep 25 15:03:51  hass[1654340]: 2023-09-25 15:03:51.178 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (81)
Sep 25 15:03:59  hass[1654340]: 2023-09-25 15:03:59.030 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (22)
Sep 25 15:10:01  hass[1654340]: 2023-09-25 15:10:01.552 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (252)
Sep 25 15:11:56  hass[1654340]: 2023-09-25 15:11:56.533 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (104)
Sep 25 15:21:09  hass[1654340]: 2023-09-25 15:21:09.729 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (10)
Sep 25 15:21:45  hass[1654340]: 2023-09-25 15:21:45.386 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (69)
Sep 25 15:34:35  hass[1654340]: 2023-09-25 15:34:35.075 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (35)
Sep 25 15:40:16  hass[1654340]: 2023-09-25 15:40:16.367 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (0)
Sep 25 15:43:42  hass[1654340]: 2023-09-25 15:43:42.466 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (81)
Sep 25 15:45:55  hass[1654340]: 2023-09-25 15:45:55.034 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (10)
Sep 25 15:47:33  hass[1654340]: 2023-09-25 15:47:33.650 WARNING (MainThread) [pyplumio.protocol] Can't process received frame: Incorrect frame checksum (0)
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by denpamusic** [comment]
Hi,

Thanks for the feedback!

_tldr: don't worry about it, those are normal for RS485. This is why checksums are there in a first place. Some more information on it below._

I can confirm that Plum only uses modified BCC for a checksum, since I have sources for the original ecoNET  300 firmware.

I also see no pattern in checksum mismatches in the log, that you provided. Checksum values themselves (listed in brackets) in this case bear no significance. The controller send frames in regular intervals, so if some frames had different checksum algo, error messages would be repeated in the same interval, that this hypothetical frame is being sent out, yet warnings in your log come at totally random intervals.

There's a reason why checksum mismatch is reported as `warning` and not an `error` and I consider lowering it even further to `info`. RS485 have no built-in flow control mechanism, it really been mostly designed for one way master-slave communication. EM controller acts as master here and constantly sending out status frames, even when PyPlumIO is trying to make some request. This leads to the collision, since we cannot really listen, when we're sending and means that some of incoming frame data gets lost and produces incorrect checksum.

This is regular occurrence when working with raw RS485 especially on lower-end hardware like raspi and in higher baudrates. Some good ICs like MAX13487 have a built-in proprietary hardware flow control for RS485. Those produce way less mismatches, but they can still occur during specific circumstances (e. g. when large enough message overflows IC's internal buffer)

**Comment by swex** [comment]
Thank you for detailed explanation! I missed the point where controller continuously sending data, even without request. Now I understand where CRC errors comes from.

**Comment by denpamusic** [comment]
> Thank you for detailed explanation!

You're welcome! If you're interested, some more information on protocol is available in [documentation](https://pyplumio.denpa.pro/protocol.html). Some of it reverse engineered by myself and community at elektroda.pl forums, some obtained from obfuscated EN300 source code.

After some consideration I've decided that now is as good of a time as any to lower logging level for checksum-related errors, since project matured enough to hide non-important stuff and keeping clean log is important for HASS users😉

Since https://github.com/denpamusic/PyPlumIO/commit/827d9804557aa3f105eea44f7737a1a579be121f, checksum-related errors will be logged at `debug` level so you shouldn't see them unless actively debugging.
